File: Models/Note/NoteModel.cs
namespace ManageLife.Models
{
    public class NoteModel
    {
        public string Id { get; set; } = default!;
        public string Title { get; set; } = default!;
        public string? Content { get; set; }
        public List<NoteTagModel> Tags { get; set; } = [];
        public DateTime CreatedTime { get; set; }
        public DateTime? UpdatedTime { get; set; }
    }
}

File: ClientSrc/Models/Note/NoteModel.ts
namespace App {
    export interface NoteTagModel { id: string; name: string; color: string; }
    export interface NoteModel {
        id: string; title: string; content: string | null;
        tags: NoteTagModel[]; createdTime: string; updatedTime: string | null;
    }
    export interface NoteDetailModel extends NoteModel {
        linkedNotes: NoteModel[];
        backlinkNotes: NoteModel[];
    }
}
